env/staging/create_metrics_lambda/lambda/create_metric.js

```diff
@@ -21,31 +21,69 @@ exports.handler = async (event, context) => {
     // expire after 24 hours
     const ttl = (Math.floor(Date.now()/1000) + 86400).toString();
 
-    const params = {
-        TableName: process.env.TABLE_NAME,
-        Item : {
-            "uuid": {
-                S: uuidv4(),
-            },
-            "expdate" : {
-                N: ttl,
-            },
-            "raw": {
-                S: event.body,
-            },
-        },
-    };
-
+    const eventBody = event.body;
+    
     try {
+        // The maximum item size in DynamoDB is 400 KB
+        const payloadLength = new TextEncoder().encode(event.body.payload).length;
+        if (payloadLength > process.env.SPLIT_THRESHOLD){
+            console.info(`Large payload being split; size: ${payloadLength}`);
+            const results = splitPayload(event.body.payload);
 
-        await dynamodb.putItem(params).promise();
+            results.forEach((result) => {
+                result.forEach((r) => {
+                    eventBody.payload = r
+                    writePayload(eventBody, ttl);
```

Comment:
Ah, I see `writePayload` awaits. Maybe move that await up here to make it more symmetrical with the else clause?

Reply:
tried moving the await and got hit with a unresolved promise warning...i'll need to fix my async await...something seems off. Might need to pair with someone since my javascript async/await is rusty

Reply:
fixed using ES6 `for..of`